prusti-tests/tests/typecheck/fail/type-model/generic-param.rs

```diff
@@ -0,0 +1,12 @@
+extern crate prusti_contracts;
+use prusti_contracts::*;
+
+use std::vec::Vec;
+
+#[model]
+struct Vec<T> { //~ ERROR: cannot find type `T` in this scope [E0412]
```

Comment:
So the generics here are the instantiations? Is it not possible to define a type model for a generic type?

Reply:
Currently it is not possible to define a model for a generic type.
It is, however, possible to define models for different concrete types (see [this test](https://github.com/vl0w/prusti-dev/blob/650ac711ca2e1af82144cdf560086e00a27954b1/prusti-tests/tests/verify/pass/type-model/models-for-different-monomorphizations.rs)).

To support generic type parameters as well, we would need annotations whether we refer to a _concrete_ or _generic_ type as we did in extern trait specs (e.g. [here](https://github.com/vl0w/prusti-dev/blob/extern-trait-specs/prusti-tests/tests/verify/pass/extern-spec/traits/generics-1.rs) or [here](https://github.com/vl0w/prusti-dev/blob/extern-trait-specs/prusti-tests/tests/verify/pass/extern-spec/traits/generics-2.rs))

Reply:
It might be necessary to add this feature, because we want to have a generic model for `std::iter::Iter`

Reply:
This feature should be implemented by [e76b3ac](https://github.com/viperproject/prusti-dev/pull/836/commits/e76b3ac6f4a0791f2601cb0a1fe340a7f78eda7e)